pkg/sql/logictest/tests/local-vec-off/local-vec-off_test: TestLogic_hash_join failed

pkg/sql/logictest/tests/local-vec-off/local-vec-off_test.TestLogic_hash_join [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/15635045?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/15635045?buildTab=artifacts#/) on master @ [908d0ee19f7e71a3d8764b306b0832e7beeedb93](https://github.com/cockroachdb/cockroach/commits/908d0ee19f7e71a3d8764b306b0832e7beeedb93):


```
=== RUN   TestLogic_hash_join
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/f37a458c473060317c1dd3f4c39280ac/logTestLogic_hash_join2632378724
    test_log_scope.go:81: use -show-logs to present logs inline
    test_server_shim.go:144: cluster virtualization disabled in global scope due to issue: #76378 (expected label: C-bug)
[23:15:39] setting distsql_workmem='16752B';
[23:15:40] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/12284/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/logictest/tests/local-vec-off/local-vec-off_test_/local-vec-off_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/hash_join with config local-vec-off: 39 tests, 0 failures
    logic.go:4135: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/12284/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/logictest/tests/local-vec-off/local-vec-off_test_/local-vec-off_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/hash_join:224: error while processing
    logic.go:4135: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/12284/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/logictest/tests/local-vec-off/local-vec-off_test_/local-vec-off_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/hash_join:224: 
        expected success, but found
        (0A000) unimplemented: can't spill column type RECORD to disk
        disk_row_container.go:158: in MakeDiskRowContainer()
    panic.go:626: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/f37a458c473060317c1dd3f4c39280ac/logTestLogic_hash_join2632378724
--- FAIL: TestLogic_hash_join (1.55s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogic_hash_join.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-39525

pkg/sql/logictest/tests/local-vec-off/local-vec-off_test.TestLogic_hash_join [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/15649613?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/15649613?buildTab=artifacts#/) on master @ [a664645d2c318ad29151cfdea49876ac67fb1fcc](https://github.com/cockroachdb/cockroach/commits/a664645d2c318ad29151cfdea49876ac67fb1fcc):


```
=== RUN   TestLogic_hash_join
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/f37a458c473060317c1dd3f4c39280ac/logTestLogic_hash_join2145978794
    test_log_scope.go:81: use -show-logs to present logs inline
    test_server_shim.go:144: cluster virtualization disabled in global scope due to issue: #76378 (expected label: C-bug)
[20:39:15] setting distsql_workmem='13246B';
[20:39:16] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/4828/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/logictest/tests/local-vec-off/local-vec-off_test_/local-vec-off_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/hash_join with config local-vec-off: 39 tests, 0 failures
    logic.go:4135: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/4828/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/logictest/tests/local-vec-off/local-vec-off_test_/local-vec-off_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/hash_join:224: error while processing
    logic.go:4135: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/4828/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/logictest/tests/local-vec-off/local-vec-off_test_/local-vec-off_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/hash_join:224: 
        expected success, but found
        (0A000) unimplemented: can't spill column type RECORD to disk
        disk_row_container.go:158: in MakeDiskRowContainer()
    panic.go:626: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/f37a458c473060317c1dd3f4c39280ac/logTestLogic_hash_join2145978794
--- FAIL: TestLogic_hash_join (1.39s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogic_hash_join.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

pkg/sql/logictest/tests/local-vec-off/local-vec-off_test.TestLogic_hash_join [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/15651374?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/15651374?buildTab=artifacts#/) on master @ [e1d779792d1126c2f1937e7072f2d9d3776a3308](https://github.com/cockroachdb/cockroach/commits/e1d779792d1126c2f1937e7072f2d9d3776a3308):


```
=== RUN   TestLogic_hash_join
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/f37a458c473060317c1dd3f4c39280ac/logTestLogic_hash_join3856552977
    test_log_scope.go:81: use -show-logs to present logs inline
    test_server_shim.go:144: cluster virtualization disabled in global scope due to issue: #76378 (expected label: C-bug)
[22:09:47] setting distsql_workmem='11009B';
[22:09:48] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/8625/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/logictest/tests/local-vec-off/local-vec-off_test_/local-vec-off_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/hash_join with config local-vec-off: 39 tests, 0 failures
    logic.go:4135: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/8625/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/logictest/tests/local-vec-off/local-vec-off_test_/local-vec-off_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/hash_join:224: error while processing
    logic.go:4135: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/8625/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/logictest/tests/local-vec-off/local-vec-off_test_/local-vec-off_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/hash_join:224: 
        expected success, but found
        (0A000) unimplemented: can't spill column type RECORD to disk
        disk_row_container.go:158: in MakeDiskRowContainer()
    panic.go:626: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/f37a458c473060317c1dd3f4c39280ac/logTestLogic_hash_join3856552977
--- FAIL: TestLogic_hash_join (1.27s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogic_hash_join.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

--- a/pkg/sql/logictest/testdata/logic_test/hash_join
+++ b/pkg/sql/logictest/testdata/logic_test/hash_join
@@ -221,5 +221,13 @@ SELECT * FROM t44797_2 WHERE EXISTS (SELECT * FROM t44797_2 AS l, t44797_3 AS r 
 statement ok
 CREATE TABLE table57696(col_table TIME NOT NULL)
 
+# Prevent the next query from spilling to disk which might error out in the
+# row-by-row engine (#125607).
+statement ok
+SET distsql_workmem = '64MiB';
+
 statement ok
 WITH cte (col_cte) AS ( SELECT * FROM ( VALUES ( ( 'false':::JSONB, '1970-01-05 16:57:40.000665+00:00':::TIMESTAMPTZ ) ) ) EXCEPT ALL SELECT * FROM ( VALUES ( ( ' [ [[true], [], {}, "b", {}], {"a": []}, {"c": 2.05750813403415} ] ':::JSONB, '1970-01-10 05:23:26.000428+00:00':::TIMESTAMPTZ ) ) ) ) SELECT * FROM cte, table57696
+
+statement ok
+RESET distsql_workmem;

